controllers/drplacementcontrol.go

```diff
@@ -462,6 +456,221 @@ func (d *DRPCInstance) getCurrentHomeClusterName(toCluster string, drClusters []
+}
+
+// checkRegionalFailoverPrerequisites checks for any RegionalDR failover prerequsites that need to be met on the
+// failoverCluster before initiating a failover.
+// Returns:
+//   - bool: Indicating if prerequisites are met
+//
+// TODO: Write envtests for this function and desendents
+func (d *DRPCInstance) checkRegionalFailoverPrerequisites() bool {
+	d.setProgression(rmn.ProgressionWaitForStorageMaintenanceActivation)
+
+	if required, activationsRequired := requiresRegionalFailoverPrequisites(
+		d.vrgs, d.instance.Spec.FailoverCluster, d.log); required {
+		for _, drCluster := range d.drClusters {
+			if drCluster.Name != d.instance.Spec.FailoverCluster {
+				continue
+			}
+
+			return checkFailoverMaintenanceActivations(drCluster, activationsRequired, d.log)
+		}
+	}
+
+	return true
+}
+
+// requiresRegionalFailoverPrequisites checks protected PVCs as reported by the last known Primary cluster
+// to determine if this instance requires failover maintenance modes to be active prior to initiating
+// a failover
+func requiresRegionalFailoverPrequisites(
+	vrgs map[string]*rmn.VolumeReplicationGroup,
+	failoverCluster string,
+	log logr.Logger,
+) (
+	bool,
+	map[string]rmn.StorageIdentifiers,
+) {
+	activationsRequired := map[string]rmn.StorageIdentifiers{}
+
+	vrg := getLastKnownPrimaryVRG(vrgs, failoverCluster)
+	if vrg == nil {
+		// TODO: Is this an error, should we ensure at least one VRG is found in the edge cases?
+		// Potentially missing VRG and so stop failover? How to recover in that case?
+		log.Info("Failed to find last known primary", "cluster", failoverCluster)
+
+		return false, activationsRequired
+	}
+
+	for _, protectedPVC := range vrg.Status.ProtectedPVCs {
+		if len(protectedPVC.StorageIdentifiers.VolumeReplicationClassModes) == 0 {
+			continue
+		}
+
+		if !hasMode(protectedPVC.StorageIdentifiers.VolumeReplicationClassModes, rmn.Failover) {
+			continue
+		}
+
+		// TODO: Assumption is that if there is a mMode then the ReplicationID is a must, err otherwise?
+		key := protectedPVC.StorageIdentifiers.CSIProvisioner + protectedPVC.StorageIdentifiers.ReplicationID
+		if _, ok := activationsRequired[key]; !ok {
+			activationsRequired[key] = protectedPVC.StorageIdentifiers
+		}
+	}
+
+	return len(activationsRequired) != 0, activationsRequired
+}
+
+// getLastKnownPrimaryVRG gets the last known Primary VRG from the cluster that is not the current targetCluster
+// This is done inspecting VRGs from the MCV reports, and in case not found, fetching it from the s3 store
+func getLastKnownPrimaryVRG(
+	vrgs map[string]*rmn.VolumeReplicationGroup,
+	targetCluster string,
+) *rmn.VolumeReplicationGroup {
+	var vrgToInspect *rmn.VolumeReplicationGroup
+
+	for drcluster, vrg := range vrgs {
+		if drcluster == targetCluster {
+			continue
+		}
+
+		if isVRGPrimary(vrg) {
+			// TODO: Potentially when there are more than on primary VRGs find the best one?
+			vrgToInspect = vrg
+
+			break
+		}
+	}
+
+	if vrgToInspect == nil {
```

Comment:
Most likely the vrgs map will be empty on failover.  That means, most of the time (in real test env) `vrgToInspect` will be nil. So getting it from the s3 store is necessary.

Reply:
Ack, TODO as per s3 comments earlier.

Reply:
Tracked in https://github.com/RamenDR/ramen/issues/835